packages/manager/modules/pci/src/projects/project/kubernetes/details/service/OidcProvider.class.js

```diff
@@ -0,0 +1,18 @@
+export default class OidcProvider {
+  /**
+   * Create a OIDC Provider instance
+   * @param oidcProviderModel {Object}: native oidcProvider JS object
+   * */
+  constructor(oidcProviderModel) {
+    Object.assign(this, oidcProviderModel);
+  }
+
+  isDefined() {
+    const { clientId, issuerUrl } = this;
+    return clientId && clientId !== '' && issuerUrl && issuerUrl !== '';
+  }
+
+  copy() {
```

Comment:
Not sure it should be this class responsibility.

Reply:
copy function, has to return make a copy of current OidcProvider instance.
I think is the right place.

Reply:
This copy function doesn't returns a deep `copy` of the original object.

Reply:
Yes it do, we will have a deep copy of object.
I think you have same remark as @radireddy here: https://github.com/ovh/manager/pull/5860#discussion_r744532871.

Reply:
Update